### PR TITLE
[CARBONDATA-1592] Added event listeners

### PR DESCRIPTION
**Pull request by ManoharVanam**
Description : Added Event Listener interface to Carbondata. This will allow extending the current functionality of various commands to perform various other operations.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by ravipesala** [comment]
retest this please

**Comment by ravipesala** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1857/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1432/

**Comment by jackylk** [comment]
@ManoharVanam please rebase

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1496/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/LoadTableCommand.scala [hunk 1]
```diff
@@ -139,6 +140,15 @@ case class LoadTableCommand(
...
         hadoopConf
       )
+      val operationContext = new OperationContext
+      val loadTablePreExecutionEvent: LoadTablePreExecutionEvent =
+        new LoadTablePreExecutionEvent(sparkSession,
+          null,
+          carbonLoadModel,
+          factPath,
+          dataFrame.isDefined,
+          optionsFinal)
```
Comment (manishgupta88) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/LoadTableCommand.scala]: carbonloadmodel is already getting passed... it there any other property that is getting used from optionsFinal variable?

**Comment by manishgupta88** [comment]
Apart from 1 comment given, LGTM

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/LoadTableCommand.scala (hunk 1 above)
Comment (ManoharVanam) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/LoadTableCommand.scala]: yeah , Iam getting bad records info from optionsFinal for validation purpose in pre event listener.
after validating we are setting to carbonLoadModel, so this is required

Comment (manishgupta88) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/LoadTableCommand.scala]: ok

**Comment by ManoharVanam** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1505/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1913/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1538/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1939/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**